src/test/run-make-fulldeps/hotplug_codegen_backend/Makefile

```diff
@@ -4,6 +4,6 @@ all:
 	/bin/echo || exit 0 # This test requires /bin/echo to exist
 	$(RUSTC) the_backend.rs --crate-name the_backend --crate-type dylib \
 		-o $(TMPDIR)/the_backend.dylib
-	$(RUSTC) some_crate.rs --crate-name some_crate --crate-type bin -o $(TMPDIR)/some_crate \
+	$(RUSTC) some_crate.rs --crate-name some_crate --crate-type lib -o $(TMPDIR)/some_crate \
 		-Z codegen-backend=$(TMPDIR)/the_backend.dylib -Z unstable-options
```

Comment:
Shouldn't we have two tests here, one for `lib` and one for `bin` ?

Reply:
Bin doesn't work without adding several lang items. `some_crate.rs` is `#![no_core]`, because `libcore` and `libstd` use `ar` based rlibs, while this pr replaces them with plain metadata for `MetadataOnlyCodegenBackend`, because it didn't support dylibs anyway.